cellmlmanip/model.py

```diff
@@ -619,6 +619,205 @@ def variables(self):
         """ Returns an iterator over this model's variable symbols. """
         return self._name_to_symbol.values()
 
+    def add_input(self, name, units, is_output=False):
+        """ Adds an additional input for the variable name to the model with new units
+        and apply any necessary conversions/add additional equations.
+
+        For example:
+        Original model
+            x = 1 [pA]
+              in [pA]
+              oxmeta: current
+            y = 1 / x
+              in [1/pA]
+
+        add_input('x', 'nA') creates model
+            input_current = 1e-3 [nA]
+              in [nA]
+              oxmeta: current
+            x = input_current * 1e3 [pA/nA]
+              in [pA]
+            y = 1 / x
+              in [1/pA]
+
+        :param name: name of variable whose units are to be changed
+        :param units: A `pint` units representation
+        :param is_output: the variable is an output rather than an input
+        """
+        # check units is of type Unit
+        assert isinstance(units, self.units.ureg.Unit)
+
+        # check variable is in model
+        original_variable = self.get_symbol_by_name(name)
+        if not original_variable:
+            raise KeyError('No variable with name "%s" found.' % name)
+
+        original_units = original_variable.units
+
+        # no conversion necessary
+        if original_units == units:
+            return
+
+        state_symbols = self.get_state_symbols()
+        free_symbol = self.get_free_variable_symbol()
+        # conversion_factor for old units to new units
+        cf = self.units.get_conversion_factor(from_unit=original_units, to_unit=units)
+
+        # create new variable and relevant equations
+        new_variable = self._convert_variable_instance(original_variable, cf, units)
+
+        # if is output do not need to do additional changes for state/free symbols
+        if is_output:
```

Comment:
Actually, I'm not at all convinced by the idea of having `is_output` as an option to `add_input`! It may be convenient for implementation, but it looks odd as an API :)

Could we instead have common internal methods that both `add_input` and `add_output` call, that give us `units`, `original_variable`, etc? (We already have one-liners to set `cf` and `new_variable`.) Then `add_output` can still be very short and the API is clearer. Things like getting a `Unit` object from either name or `Unit`, getting a `VariableDummy` from object or annotation, etc are probably generally useful in a few places too.